index.ts

```diff
@@ -179,30 +192,21 @@ wss.on("request", function (request) {
     };
   }
   const room: Room = rooms[roomId];
+  let conn: Connection = {
+    name,
+    connection,
+    id
+  };
+  room.connections.push(conn);
 
   connection.on("message", function (message) {
-    let name = "";
-    let id = "";
-
-    room.connections
-      .filter((item) => item.connection == connection)
-      .map((item) => {
-        name = item.name;
-        id = item.id;
-      });
-
     let data = JSON.parse(message.utf8Data);
     // console.log(`Inbound: ${message.utf8Data}`);
     let card: Card;
     let msg: string;
 
     switch (data.type) {
       case "join":
-        do {
-          id = generateId();
-        } while (room.connections.filter((item) => item.id == id).length > 0);
-
-        room.connections.push({ name: data.name, connection: connection, id: id });
```

Comment:
This is removed here but re-introduced in the very next commit. Reworking this and the next commit could reduce both diffs